src/fields/multi_field.impl.hh

```diff
@@ -377,10 +377,15 @@ bool MultiField<spacedim, Value>::MultiFieldFactory::is_active_field_descriptor(
 
 
 template<int spacedim, class Value>
-void MultiField<spacedim, Value>::set_dependency(FieldSet &field_set) {
+std::vector<const FieldCommon *> MultiField<spacedim, Value>::set_dependency(FieldSet &field_set, unsigned int i_reg) {
+    std::vector<const FieldCommon *> depend_fields;
     for(unsigned int i_comp=0; i_comp < this->shared_->comp_names_.size(); i_comp++) {
-        sub_fields_[i_comp].set_dependency(field_set);
+        auto add = sub_fields_[i_comp].set_dependency(field_set, i_reg);
+        depend_fields.insert(depend_fields.end(), add.begin(), add.end());
     }
+    std::sort(depend_fields.begin(), depend_fields.end());
```

Comment:
No need to making the list unique, DFS should work even for that, as long as we have visited fields marked.